client/INLClient.py

```
@@ -80,28 +94,106 @@ def check_settings(self):
+
+        self.create_build_root()
+        self.remove_build_root()
+
+    def get_build_root(self):
+        """
+        Gets the build root.
+        Raises:
+          BaseClient.ClientException: If the BUILD_ROOT env variable is not set.
+        """
+        build_root = os.getenv('BUILD_ROOT', None)
+        if build_root is None:
+            logger.error("Failed to get 'BUILD_ROOT' environment variable")
+            raise BaseClient.ClientException('Faild to get BUILD_ROOT; varaible not set')
+        return build_root
+
+    def build_root_exists(self):
+        """
+        Returns:
+          True if the BUILD_ROOT exists, False otherwise
+        """
+        return os.path.isdir(self.get_build_root())
+
+    def remove_build_root(self):
+        """
+        Removes the build root.
+        Raises:
+          BaseClient.ClientException: If BUILD_ROOT does not exist, or the directory removal failed.
+        """
+        build_root = self.get_build_root()
+
+        if self.build_root_exists():
+            logger.info('Removing BUILD_ROOT {}'.format(build_root))
+            shutil.rmtree(build_root)
+        else:
+            raise BaseClient.ClientException('Failed to remove BUILD_ROOT {}; it does not exist'.format(build_root))
 
-    def run(self, single=False):
+    def create_build_root(self):
+        """
+        Creates the build root.
+        Raises:
+          BaseClient.ClientException: If the BUILD_ROOT directory could not be created or if it already exists.
+        """
+        build_root = self.get_build_root()
+        if self.build_root_exists():
+            raise BaseClient.ClientException('Failed to create BUILD_ROOT {}; it already exists'.format(build_root))
+        else:
+            try:
+                os.mkdir(build_root)
+                logger.info('Created BUILD_ROOT {}'.format(build_root))
+            except:
+                raise BaseClient.ClientException('Failed to create BUILD_ROOT {}'.format(build_root))
```

This hides the exception that is causing the problem, which is often very confusing to figure out what happened.
```suggestion
            try:
                os.mkdir(build_root)
                logger.info('Created BUILD_ROOT {}'.format(build_root))
            except:
                logger.exception(Failed to create BUILD_ROOT {}'.format(build_root))
                raise
```